java/client/src/org/openqa/selenium/support/pagefactory/ByChained.java

```diff
@@ -48,10 +48,21 @@ public ByChained(By... bys) {
 
   @Override
   public WebElement findElement(SearchContext context) {
-    List<WebElement> elements = findElements(context);
-    if (elements.isEmpty())
-      throw new NoSuchElementException("Cannot locate an element using " + toString());
-    return elements.get(0);
+    if (bys.length == 0) {
+      throw new NoSuchElementException("No Bys were specified in this " + getClass().getSimpleName());
+    }
+
+    WebElement elem = null;
+    for (By by : bys) {
+      boolean isFirstIteration = elem == null;
+      if (isFirstIteration) {
+        elem = by.findElement(context);
+      } else {
+        elem = elem.findElement(by);
+      }
+    }
+    assert elem != null; // NoSuchElement should have already been thrown
```

Comment:
I was trying to point out the previous code path used by.findElements <- s is important and would continue searching through all by's looking for elements.

Your change makes it so if the first by does not locate an element (or essentially any by in the list) it would throw a not found exception.

The change I proposed keeps the logic in place where it would end up calling all the by's in the chain. 

But the existing code has some odd edge cases by the artifact of the last by in the chain is the only one it really cares about and it can be hard to predict results if intermediate by's don't return any elements.